core/src/main/java/jenkins/model/lazy/LazyBuildMixIn.java

```diff
@@ -265,6 +268,29 @@ public final RunT getNearestOldBuild(int n) {
         return builds.search(n, AbstractLazyLoadRunMap.Direction.DESC);
     }
 
+    /**
+     * Suitable for {@link Job#getEstimatedDurationCandidates}.
+     * @since TODO
+     */
+    public List<RunT> getEstimatedDurationCandidates() {
+        var loadedBuilds = builds.getLoadedBuilds().values(); // reverse chronological order
+        List<RunT> candidates = new ArrayList<>(3);
+        for (Result threshold : List.of(Result.UNSTABLE, Result.FAILURE)) {
+            for (RunT build : loadedBuilds) {
+                if (!build.isBuilding()) {
+                    Result result = build.getResult();
+                    if (result != null && result.isBetterOrEqualTo(threshold)) {
```

Comment:
@jglick am I missing something here?
`Result threshold : List.of(Result.UNSTABLE, Result.FAILURE` 

if there are only 2 loaded builds (both success) (or 5 loaded builds but only 2 of which are successful)  won't this cause the loop to add both of them for the `UNSTABLE` case and then add one duplicate for the FAILURE case?

Should the `List<RunT>` be a `Set` instead and only switched to a List at the end.  (not sure why a list and not a `SortedSet` - as duplicate Runs don;t make sense here).

Reply:
Possibly. I will check soon if there is existing test coverage specifically for this method, and see if I can reproduce any issues like that.

Reply:
#8233. Sorry for the delay.